### ISSUE TITLE
docker:build remove <none> container

### ISSUE THREAD

**Issue by benathmane** [fabric8io/docker-maven-plugin#501]
Hello,

what's the argument to remove or not creat <none> container ?
My command line : mvn clean package install docker:build
Thanks

**Comment by rhuss** [comment]
Sorry, didn't get question. What is your issue ?

**Comment by benathmane** [comment]
I want to know if there's an argument (not a command line) to remove <none> containers after a mvn clean package install docker:build :)

**Comment by rhuss** [comment]
`docker:build` doesnt create containers, only images. 

`docker:start` creates containers and they can be removed at `docker:stop` after running with the option `keepContainer=false`. There is also a `docker:remove` goal for removing images.

Please let me know when and where the manual is not clear enough so that I can fix that.

**Comment by benathmane** [comment]
Sorry,
after this command : "mvn clean package install docker:build"

I have : 
hygieia-api                           latest              cb5a97d238b5        4 minutes ago       708 MB
<none>                                <none>              c117c90f967d        47 minutes ago      700.8 MB
<none>                                <none>              b7ba55168775        47 minutes ago      700.8 MB
<none>                                <none>              d98bc12b591c        47 minutes ago      700.1 MB
<none>                                <none>              fc7b12aa101b        47 minutes ago      699.3 MB
<none>                                <none>              3834b5446f96        47 minutes ago      708 MB

I do not understand why it creates <none> images :(

**Comment by rhuss** [comment]
These are different layers which are stacked (and the size is sum of this layer + all layers below). That is a normal Docker feature (layered filesystem), you can remove any of the layer without destroying the whole image.

**Comment by benathmane** [comment]
Yes with this command : docker images|sed "1 d"|grep \<none\>|awk '{print $3}'|sort|uniq|xargs docker rmi

thanks :)

**Comment by rhuss** [comment]
Ok, so you mean to clean up images from a *previous* build right ?

BTW, the easiest way to remove dangling images is `docker rmi $(docker images -qa -f 'dangling=true')`

**Comment by benathmane** [comment]
Thanks :+1:

**Comment by rhuss** [comment]
Yeah, might make sense that if an image with the given name already exists when building that it could be (optionally) deleted.

**Comment by jgangemi** [comment]
the plugin already cleans up images from a previous build provided that image is still properly tagged/labeled.

**Comment by idelix** [comment]
I face the same problem after building and pushing image with same name and tag previous one is not getting deleted, wonder if there is a way for plugin to do it automatically. Otherwise the command provided by rhuss is the option docker rmi $(docker images -qa -f 'dangling=true')

mvn package dockerfile:build dockerfile:push

REPOSITORY                                  TAG                 IMAGE ID            CREATED             SIZE
servername.azurecr.io/gs-spring-boot-docker   latest              2a356c654697        23 seconds ago      119MB
servername.azurecr.io/gs-spring-boot-docker   <none>              f8d0d3b1dd60        39 minutes ago      119MB
openjdk                                     8-jdk-alpine        224765a6bdbe        2 months ago        102MB

**Comment by SephReed** [comment]
> docker rmi $(docker images -qa -f 'dangling=true')

wtf?  Is this really the best way to not fill of a `.qcow` with versions of a build you don't care about?  Can't you just overwrite the old one or something?  Wouldn't this kind of feature fall under "basic functionality" rather than something you have to do by hand?

**Comment by ulrichmeyer** [comment]
Execute "docker image prune" to delete old versions
See details on: https://stackoverflow.com/questions/45142528/what-is-a-dangling-image-and-what-is-an-unused-image

**Comment by grofte** [comment]
I can easily
```
docker image rm $(DOCKER_IMAGE)
docker build -t $(DOCKER_IMAGE) .
```
in a bash or make script but if the image doesn't exist then everything comes crashing down. Is there no option to delete an image **if it exists**?

**Comment by szabikr** [comment]
thanks @grofte, good idea.

However, if there's a container that runs and has been created from the same image then docker won't allow you to delete the image.

**Comment by grofte** [comment]
> thanks @grofte, good idea.
> 
> However, if there's a container that runs and has been created from the same image then docker won't allow you to delete the image.

I usually use `docker run --rm=true` which closes down the container after use. I don't exactly know why it isn't default. You could also force the deletion but I don't know if I want to automate that. There might be a reason why the container is still running.

**Comment by J3ronimo** [comment]
I'm running 

```
docker rmi myimage
docker build -t myimage .
```

in a batch (Docker Desktop on Windows) and I also feel like this should be doable in a single command, ignoring the initial remove if the image doesn't exist.

**Comment by dpresbit** [comment]
docker system prune
